packages/swh-plugins.lv2/PKGBUILD

```diff
@@ -0,0 +1,39 @@
+# Maintainer: OSAMC <https://github.com/osam-cologne/archlinux-proaudio>
+# Contributor: Florian Hülsmann <fh@cbix.de>
+
+_name=lv2
+pkgname=swh-plugins.lv2
+pkgver=1.0.16
+pkgrel=1
+pkgdesc="LV2 port of Steve Harris' plugins suite"
+arch=(x86_64 aarch64)
+url='http://plugin.org.uk/'
+license=(GPL3)
+groups=(lv2-plugins pro-audio)
+depends=()
+makedepends=(fftw libxslt)
+checkdepends=(kxstudio-lv2-extensions lilv lv2lint)
+optdepends=('lv2-host: for running the plugins')
+provides=(swh-plugins)
+source=("$pkgname-$pkgver.tar.gz::https://github.com/swh/$_name/archive/refs/tags/v$pkgver.tar.gz")
+sha256sums=('bc24512de6e2fb7a493226e2e01a80ba8462a318b15c3b0fd0cd914b018c3548')
+
+build() {
+  cd $_name-$pkgver
+  make
+}
+
+check() {
+  _lv2path="$srcdir"/$_name-$pkgver/plugins
+  for _plugin in $(LV2_PATH="$_lv2path" lv2ls 2>/dev/null); do
```

Comment:
Setting `LV2_PATH` works quite well, but I have to `2>/dev/null` these errors from liblilv because apparently some of the plugins are not listed as compile targets. We might eventually patch the Makefile to enable these as well, but maybe there's a reason for them not being included.
```
error: failed to open file /home/florian/build/swh-lv2/plugins/debug-swh.lv2/manifest.ttl (No such file or directory)
lilv_world_load_file(): error: Error loading file `file:///home/florian/build/swh-lv2/plugins/debug-swh.lv2/manifest.ttl'
lilv_world_load_bundle(): error: Error reading file:///home/florian/build/swh-lv2/plugins/debug-swh.lv2/manifest.ttl
error: failed to open file /home/florian/build/swh-lv2/plugins/imp-swh.lv2/manifest.ttl (No such file or directory)
lilv_world_load_file(): error: Error loading file `file:///home/florian/build/swh-lv2/plugins/imp-swh.lv2/manifest.ttl'
lilv_world_load_bundle(): error: Error reading file:///home/florian/build/swh-lv2/plugins/imp-swh.lv2/manifest.ttl
error: failed to open file /home/florian/build/swh-lv2/plugins/gsm-swh.lv2/manifest.ttl (No such file or directory)
lilv_world_load_file(): error: Error loading file `file:///home/florian/build/swh-lv2/plugins/gsm-swh.lv2/manifest.ttl'
lilv_world_load_bundle(): error: Error reading file:///home/florian/build/swh-lv2/plugins/gsm-swh.lv2/manifest.ttl
error: failed to open file /home/florian/build/swh-lv2/plugins/step_muxer-swh.lv2/manifest.ttl (No such file or directory)
lilv_world_load_file(): error: Error loading file `file:///home/florian/build/swh-lv2/plugins/step_muxer-swh.lv2/manifest.ttl'
lilv_world_load_bundle(): error: Error reading file:///home/florian/build/swh-lv2/plugins/step_muxer-swh.lv2/manifest.ttl
error: failed to open file /home/florian/build/swh-lv2/plugins/notch_iir-swh.lv2/manifest.ttl (No such file or directory)
lilv_world_load_file(): error: Error loading file `file:///home/florian/build/swh-lv2/plugins/notch_iir-swh.lv2/manifest.ttl'
lilv_world_load_bundle(): error: Error reading file:///home/florian/build/swh-lv2/plugins/notch_iir-swh.lv2/manifest.ttl
```

Reply:
```sh
source=("$pkgname-$pkgver.tar.gz::https://github.com/swh/$_name/archive/refs/tags/v$pkgver.tar.gz"
        'valgrind-dpf.supp')
sha256sums=('bc24512de6e2fb7a493226e2e01a80ba8462a318b15c3b0fd0cd914b018c3548'
            '414a5be1b395938754f697eca336f160a190d6ff7ec5edf1fd397d0470085bc4')

[...]
check() {
  cd $_name-$pkgver
  for _dir in plugins/*.lv2; do
    if [ ! -e "$_dir/manifest.ttl" ]; then
      rm -rf "$_dir"
    fi
  done

  for _ttl in plugins/*/manifest.ttl; do
    grep 'a :Plugin' $_ttl | sed -Ee 's|^swh:(.*) a :Plugin ;|http://plugin.org.uk/swh-plugins/\1|g' | \
    while read _plugin; do
      echo "Checking plugin $_plugin with lv2lint..."
      lv2lint -q -Mpack -t 'Plugin Version *' -I `pwd`/${_ttl%/*} "$_plugin" || \
        echo "Ignoring lv2lint errors (known to fail)"
      LV2_PATH=`pwd`/plugins:/usr/lib/lv2 CARLA_BRIDGE_DUMMY=1 CARLA_BRIDGE_TESTING=native \
        valgrind \
          --error-exitcode=255 \
          --leak-check=full \
          --track-origins=yes \
          --suppressions="$srcdir"/valgrind-dpf.supp \
          /usr/lib/carla/carla-bridge-native lv2 "" \
          "$_plugin" || \
        echo "Ignoring valgrind errors (known to fail)"
    done
  done
}
```

(Source: https://github.com/DISTRHO/DPF-Plugins/blob/master/.github/workflows/build.yml#L420)

Segfaults on a couple of plugins.

Reply:
> Segfaults on a couple of plugins.

Yep, fixing them right now :)

Reply:
1577c6c1fa32ed912dd93c4a598b2c9426ce411a

Reply:
Yea, memleak checking is also a good idea for testing in general, however I believe that should not happen during package build.

Reply:
See also existing upstream tickets about segfaults.

Reply:
All segfaults fixed now 🎉